react/components/ChildrenContainer.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+import { Container } from 'vtex.store-components'
+
+import { itemPropType } from '../propTypes'
+import ChildLink from './ChildLink'
+import categoryMenuPosition, {
+  getMenuPositionValues,
+} from '../utils/categoryMenuPosition'
+
+import categoryMenu from '../categoryMenu.css'
+
+/**
+ * Component responsible for rendering an array of categories and its respective subcategories
+ */
+export default class ChildrenContainer extends Component {
+  static propTypes = {
+    /** Category to be displayed */
+    items: PropTypes.arrayOf(itemPropType).isRequired,
+    /** Department slug */
+    parentSlug: PropTypes.string,
+    /** Close menu callback */
+    onCloseMenu: PropTypes.func.isRequired,
+    /** Indicates if the children must be shown */
+    isShowing: PropTypes.bool,
+    /** Ref to the parent container. Used to position the children below it */
+    containerRef: PropTypes.object,
+    /** Whether to show second level links or not */
+    showSecondLevel: PropTypes.bool,
+    /** Defines the position of the category menu */
+    menuPosition: PropTypes.oneOf(getMenuPositionValues()),
+    /** Custom styles to item container */
+    containerStyle: PropTypes.object,
+  }
+
+  shouldRenderSecondLevel({ children }) {
+    const { showSecondLevel } = this.props
+    return children && !!children.length && showSecondLevel
+  }
+
+  render() {
+    const {
+      items,
+      menuPosition,
+      containerRef,
+      isShowing,
+      onCloseMenu,
+    } = this.props
+
+    const containerStyle = {
+      top: containerRef && containerRef.offsetTop + containerRef.clientHeight,
+      display: isShowing ? 'flex' : 'none',
```

Comment:
Would not look better withClassnames using tachyons? Like -> `isShowing ? 'flex' : 'dn'`

Reply:
Yes! Thanks for the suggestion